### PR TITLE
[BT-673]: Add PanelProgress component.

### PR DESCRIPTION
**Pull request by cjcenizal**

### CODE REVIEW

**Comment by jenslonkowski** [comment]
Looks good! But according to the examples in https://github.com/SmaatoUI/loading-indicator-prototypes the loading indicator for the lists should be smaller. Maybe we can adjust the Progress component to support this through a property.

**Comment by ahollesen** [comment]
Agree with Jens

File: src/framework/body/BodyPanelItem.spec.jsx [hunk 1]
```diff
@@ -25,12 +22,9 @@ describe('BodyPanelItem', () => {
...
 
-          const testCase = TestCaseFactory.createFromFunction(
-            BodyPanelItem,
-            props
-          );
-          expect(testCase.dom.getAttribute('class')
-            .indexOf('bodyPanelItem--largeRhythm') !== -1).toBe(true);
+          const testCase = TestCaseFactory.create(BodyPanelItem, props);
+          expect(testCase.dom.className)
+            .toContain('bodyPanelItem--largeRhythm');
```
Comment (cjcenizal) [inline src/framework/body/BodyPanelItem.spec.jsx]: I noticed these tests used an older convention, so I updated them.

File: src/guide/components/navigation/_navigation.scss [hunk 2]
```diff
@@ -9,7 +9,7 @@
   background: #7b8396;
   width: 400px;
   overflow-y: auto;
-  transition: transform 0.5s ease-out;
+  transition: transform 0.25s ease-out;
```
Comment (cjcenizal) [inline src/guide/components/navigation/_navigation.scss]: This was moving too slowly for me.

**Comment by cjcenizal** [comment]
@jenslonkowski @ahollesen Good catch guys! Please re-review.

**Comment by ahollesen** [comment]
Yup 👍

**Comment by jenslonkowski** [comment]
Great! 👍

**Pull request merged**